diff --git a/Program.cs b/Program.cs
index caaa454..31ac955 100644
--- a/Program.cs
+++ b/Program.cs
@@ -7,6 +7,7 @@ using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.Logging;
+using App.Metrics.AspNetCore;
 
 namespace WebApp1
 {
@@ -19,6 +20,9 @@ namespace WebApp1
 
         public static IWebHost BuildWebHost(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
+                .UseMetrics()
+                .UseMetricsWebTracking()
+                .UseResourceMetrics()
                 .UseStartup<Startup>()
                 .Build();
     }
diff --git a/Startup.cs b/Startup.cs
index a9775dc..a556a4e 100644
--- a/Startup.cs
+++ b/Startup.cs
@@ -8,6 +8,7 @@ using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
+using Microsoft.AspNetCore.Mvc;
 
 namespace WebApp1
 {
@@ -23,7 +24,7 @@ namespace WebApp1
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddMvc();
+            services.AddMvc(options => options.AddMetricsResourceFilter());
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
diff --git a/WebApp1.csproj b/WebApp1.csproj
index 60ee45b..5d35d42 100644
--- a/WebApp1.csproj
+++ b/WebApp1.csproj
@@ -16,4 +16,9 @@
     <DotNetCliToolReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Tools" Version="2.0.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Include="App.Metrics.AspNetCore.Mvc" Version="2.0.0-alpha" />
+    <PackageReference Include="App.Metrics.AspNetCore.Tracking" Version="2.0.0-alpha" />
+    <PackageReference Include="ResourceMetrics" Version="1.0.0" />
+  </ItemGroup>
 </Project>
